cli/create_server_reinstall_operation.go

```diff
@@ -26,110 +24,86 @@ func makeOperationServerReinstallCmd() (*cobra.Command, error) {
+			Required:    false,
+			Options:     server.SupportedOperatingSystems,
 		},
-		{
-			Name:             "hostname",
-			Description:      "The server hostname",
-			DefaultValue:     "",
-			Type:             "string",
-			RequestParamType: cmdflag.BodyParam,
-			Required:         false,
+		&cmdflag.String{
+			Name:        "Hostname",
+			Label:       "",
+			Description: "The server hostname",
+			Required:    false,
 		},
-		{
-			Name:             "ssh_keys",
-			Description:      "The SSH Keys to set on the server",
-			DefaultValue:     []string{},
-			Type:             "stringSlice",
-			RequestParamType: cmdflag.BodyParam,
-			Required:         false,
+		&cmdflag.StringSlice{
+			Name:        "ssh_keys",
+			Label:       "SSH Keys",
+			Description: "The SSH Keys to set on the server",
+			Required:    false,
 		},
-		{
-			Name:             "user_data",
-			Description:      "User data to set on the server",
-			DefaultValue:     int64(0),
-			Type:             "int64",
-			RequestParamType: cmdflag.BodyParam,
-			Required:         false,
+		&cmdflag.Int64{
+			Name:        "user_data",
+			Label:       "User Data",
+			Description: "User data to set on the server",
+			Required:    false,
 		},
-		{
-			Name:             "raid",
-			Description:      `Enum: ["raid-0","raid-1"]. RAID mode for the server`,
-			DefaultValue:     "",
-			Type:             "string",
-			RequestParamType: cmdflag.BodyParam,
-			Required:         false,
+		&cmdflag.String{
+			Name:        "raid",
+			Label:       "RAID Level",
+			Description: `Enum: ["raid-0","raid-1"]. RAID mode for the server`,
+			Required:    false,
+			Options:     server.SupportedRAIDLevels,
 		},
-		{
-			Name:             "ipxe_url",
-			Description:      "URL where iPXE script is stored on, necessary for custom image deployments.This attribute is required when iPXE is selected as operating system.",
-			DefaultValue:     "",
-			Type:             "string",
-			RequestParamType: cmdflag.BodyParam,
-			Required:         false,
+		&cmdflag.String{
+			Name:        "ipxe_url",
+			Label:       "iPXE URL",
+			Description: "URL where iPXE script is stored on, necessary for custom image deployments.This attribute is required when iPXE is selected as operating system.",
+			Required:    false,
 		},
 	}
 
-	o.Flags.Register(schema)
-}
+	o.BodyAttributesFlags.Register(bodyAttributesFlagsSchema)
+	o.PathParamFlags.Register(pathParamsFlagsSchema)
 
-func (o *CreateServerReinstallOperation) GetFlags() cmdflag.Flags {
-	return o.Flags
 }
 
-func (o *CreateServerReinstallOperation) PromptQueryParams(params interface{}) {
-	p := prompt.New(
-		prompt.NewInputText("id", "Server ID"),
-	)
-
-	p.Run(params)
+func (o *CreateServerReinstallOperation) preRun(cmd *cobra.Command, args []string) {
```

Comment:
This was the fastest way I found to implement preRun functions given our current design. I split the PathParamFlags and BodyAttributesFlags into different instances just because their `AssignValues` functions expects different arguments (interfaces), but I believe there are smarter ways to implement that. Any suggestions are more than welcome!